### PR TITLE
Add tests for registration

### PR DESCRIPTION
**Pull request by Tyil**

### CODE REVIEW

File: tests/RegistrationTest.php [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                'password' => 'test',
+                'password-verify' => 'test',
+                'accept-tos' => true,
+            ])
+        );
+    }
+
+    public function testUnmatchedPassword()
+    {
+        $this->assertRedirectedToRoute('user.create', [], [],
```
Comment (KingNoosh) [inline tests/RegistrationTest.php]: Would it not be a good idea to check what the exception here is too alongside the redirect?

File: tests/RegistrationTest.php [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                'domain' => $this->domain,
+                'password' => 'test',
+                'password-verify' => 'test',
+            ])
+        );
+    }
+
+    public function testUnknownDomain()
+    {
+        $this->assertRedirectedToRoute('user.create', [], [],
```
Comment (KingNoosh) [inline tests/RegistrationTest.php]: Would it not be a good idea to check what the exception here is too alongside the redirect?

File: tests/RegistrationTest.php [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+                    'password-verify' => 'test',
+                    'accept-tos' => true,
+                ])
+            );
+        }
+    }
+
+    public function testUncheckedTos()
+    {
+        $this->assertRedirectedToRoute('user.create', [], [],
```
Comment (KingNoosh) [inline tests/RegistrationTest.php]: Would it not be a good idea to check what the exception here is too alongside the redirect?

**Review by KingNoosh — COMMENTED** [review]
The only thing I can just see is that the unit tests might not be strong enough. However, I don't know whether this might be enough group to request changes, so I'm leaving them up to you as suggestions.

**Review by KingNoosh — APPROVED** [review]
Forgot to click approve. :^)

File: tests/RegistrationTest.php (hunk 3 above)
Comment (Tyil) [reply tests/RegistrationTest.php]: That would probably be good, yeah. I'm only checking the bare technical basics here, I'll read up on the Laravel docs to see if I can check for the specific errors given in a request. If anyone else knows about this already, do inform me.

**Review by KingNoosh — CHANGES_REQUESTED** [review]
Prepping for re-review as the Laravel update has had some breaking changes. Reply to this comment once it's at a stage where it needs reviewing. :)

**Comment by Tyil** [comment]
@KingNoosh: that was a failing seedscript :(

**Review by KingNoosh — APPROVED** [review]
Looks fine to me

**Pull request merged**